pkg/agent/proxy/endpointslicecache.go

```diff
@@ -0,0 +1,364 @@
+/*
```

Comment:
@weiqiangt I think you mentioned in the original PR that this file couldn't be copied to `third_party/`. Was that for import reasons? Maybe worth documenting as a comment in case someone wonders about it later.

Reply:
I would agree!

Also thinking should we add comments for the changes we made to the original copied files, e.g. at the function level? Or there might be too many changes to track in this way?

Reply:
If this file is located in `third_party/`,  there will be an import cycle issue when build Antrea.
```
#13 2.800 GOOS=linux go build -o /antrea/bin  -ldflags ' -X github.com/vmware-tanzu/antrea/pkg/version.Version=v0.13.0-dev -X github.com/vmware-tanzu/antrea/pkg/version.GitSHA=15f660eb -X github.com/vmware-tanzu/antrea/pkg/version.GitTreeState=dirty -X github.com/vmware-tanzu/antrea/pkg/version.ReleaseStatus=unreleased' github.com/vmware-tanzu/antrea/cmd/antrea-agent
#13 5.801 package github.com/vmware-tanzu/antrea/cmd/antrea-agent
#13 5.801       imports github.com/vmware-tanzu/antrea/pkg/agent/proxy
#13 5.801       imports github.com/vmware-tanzu/antrea/third_party
#13 5.801       imports github.com/vmware-tanzu/antrea/pkg/agent/proxy: import cycle not allowed
#13 5.811 make: *** [Makefile:31: antrea-agent] Error 1
```

Reply:
Is it possible for this?

> Also thinking should we add comments for the changes we made to the original copied files, e.g. at the function level? Or there might be too many changes to track in this way?

@weiqiangt

Reply:
Or we can put some descriptions of the changes in third_party/proxy/README.

I am not saying we must do this in this PR.